File: assets/handlers/python/invoice_create/src/create_invoice/create.py
from create_invoice import db, unmarshal


async def create_invoice(json_data: bytes) -> str:
    invoice = unmarshal.unmarshal_invoice(json_data)
    insert_dict = invoice.model_dump()
    insert_result = await db.collection.insert_one(insert_dict)
    return str(insert_result.inserted_id)

File: assets/handlers/go/invoice_create/src/create/create.go
package create

import (
	"context"
	"invoice_create/src/db"
	"invoice_create/src/models"
	"invoice_create/src/unmarshal"
)

// Creates invoice for the given invoice data.
func CreateInvoice(jsonData []byte) (string, error) {
	invoiceStruct, err := unmarshal.UnmarshalInvoice(jsonData)
	if err != nil {
		return "", err
	}

	invoice, err := models.NewInvoice(invoiceStruct)
	if err != nil {
		return "", err
	}

	insertResult, err := db.InvoiceCollection.InsertOne(context.Background(), invoice)
	if err != nil {
		return "", err
	}

	return insertResult.InsertedID.(string), nil
}
